packages/graphback/src/GraphQLBackendCreator.ts

```diff
@@ -21,17 +29,22 @@ import { logger } from './utils/logger'
 // Datamigration should be the component on it's own
 // TODO rename to backend
 export class GraphQLBackendCreator {
-
   private dataLayerManager: IDataLayerResourcesManager;
   private dbContextProvider: DatabaseContextProvider;
-  private inputContext: Type[]
+  private schemaContext: ISchemaProvider;
+  private inputContext: Type[];
+  private newSchema: string;
+  private oldSchema: string;
 
   /**
    * @param graphQLSchema string containing graphql types
    * @param config configuration for backend generator
    */
-  constructor(graphQLSchema: string, config: GraphQLGeneratorConfig) {
-    this.inputContext = createInputContext(graphQLSchema, config)
+  constructor(schemaContext: ISchemaProvider, config: GraphQLGeneratorConfig) {
+    this.schemaContext = schemaContext;
+    this.oldSchema = schemaContext.getOldSchemaText();
+    this.newSchema = schemaContext.getNewSchemaText();
```

Comment:
How about using `current` and `previousVersion`? 
Saying your schema is OLD sounds funny :)